### PR TITLE
Change to add additional properties to hbase-site.xml to enable customizations

### PR DESCRIPTION
**Pull request by bijugs**
- Change to enable customizing HBase bucketcache bucket sizes -> Property added hbase.bucketcache.bucket.sizes.
- hbase.fs.tmp.dir property added to handle situations descripbed in [HBASE-15129](https://issues.apache.org/jira/browse/HBASE-15129)

### CODE REVIEW

File: cookbooks/bcpc-hadoop/attributes/hbase.rb [hunk 1]
```diff
@@ -71,6 +73,7 @@
 default[:bcpc][:hadoop][:hbase][:site_xml].tap do |site_xml|
   site_xml['hbase.rootdir'] = "#{node['bcpc']['hadoop']['hbase']['root_dir']}"
   site_xml['hbase.bulkload.staging.dir'] = "#{node['bcpc']['hadoop']['hbase']['bulkload_staging_dir']}"
+  site_xml['hbase.fs.tmp.dir'] = "#{node['bcpc']['hadoop']['hbase']['fs_tmp_dir']}"
```
Comment (cbaenziger) [inline cookbooks/bcpc-hadoop/attributes/hbase.rb]: Please don't create vanity names for site.xml variables; simply set the site.xml variable.

**Comment by cbaenziger** [comment]
Hi Biju, thanks for this PR. Is there any value to force users to learn a "chef-bach" vocabulary of HBase tunings? Ideally, I would prefer us to use the same names as `hbase-site.xml` to avoid confusion renaming settings already agreed upon by the HBase community.

File: cookbooks/bcpc-hadoop/attributes/hbase.rb (hunk 1 above)
Comment (bijugs) [reply cookbooks/bcpc-hadoop/attributes/hbase.rb]: @cbaenziger  the change was done this way to be consistent with what seems to be a standard followed in the code historically. We can make a break with this change.

Comment (cbaenziger) [reply cookbooks/bcpc-hadoop/attributes/hbase.rb]: Hi Biju, yes, please continue to move away from these vanity names. We are working to unify on the HBase community names which allows us to use the variables directly in the `site_xml` hash. Thanks!

**Comment by bijugs** [comment]
Updated PR to set ``hbase.fs.tmp.dir`` property directly in ``site_xml`` hash.

```
site_xml['hbase.fs.tmp.dir'] = "/user/${user.name}/hbase-staging"
```

**Comment by bijugs** [comment]
Hi @cbaenziger , @leochen4891  can you have a look and let me know if further changes are required.

File: cookbooks/bcpc-hadoop/recipes/hbase_config.rb [hunk 2]
```diff
@@ -125,6 +125,9 @@
   generated_values['hbase.bucketcache.size'] = bucketcache_size
   generated_values['hbase.bucketcache.ioengine '] = node["bcpc"]["hadoop"]["hbase"]["bucketcache"]["ioengine"]
   generated_values['hbase.bucketcache.combinedcache.enabled'] = true
+  if !node["bcpc"]["hadoop"]["hbase"]["bucketcache.bucket.sizes"].nil?
+    generated_values['hbase.bucketcache.bucket.sizes'] = node["bcpc"]["hadoop"]["hbase"]["bucketcache"]["bucket"]["sizes"]
```
Comment (leochen4891) [inline cookbooks/bcpc-hadoop/recipes/hbase_config.rb]: Should it be the following?
`+    generated_values['hbase.bucketcache.bucket.sizes'] = node["bcpc"]["hadoop"]["hbase"]["bucketcache.bucket.sizes"]`

**Comment by bijugs** [comment]
Thanks @leochen4891 . Updated the PR.

**Comment by cbaenziger** [comment]
One nit but it looks like rubocop/foodcritic best practices weren't followed for the new lines? (Particularly quotes look like they'd flag.)

**Pull request merged**